util/design/gen-otp-immutable-rom-ext-json.py

```diff
@@ -86,6 +112,22 @@ def update_json_with_immutable_rom_ext_section_data(self):
         self.insert_key_value(_SIZE_FIELD_NAME, f"{hex(self.size_in_bytes)}")
         self.insert_key_value(_HASH_FIELD_NAME, f"0x{self.hash.hex()}")
 
+    def immutable_rom_ext_enable(self, hardened_true_value: int):
+        """Checks if immutable ROM extension is enabled.
+
+        This method retrieves the value of the enable field from the OTP
+        partition and compares it with the provided hardened true value.
+
+        Args:
+            hardened_true_value: The expected value for the enable field
+                                when immutable ROM extension is enabled.
+
+        Returns:
+            True if immutable ROM extension is enabled, False otherwise.
+        """
+        immutable_rom_ext_en_value = int(self.get_key_value(_ENABLE_FIELD_NAME), 0)
```

Comment:
```suggestion
        immutable_rom_ext_en_value = int(self.json_data["partitions"][_OTP_PARTITION_NAME].get(_ENABLE_FIELD_NAME, 0))
```
should also work I believe.

Reply:
`self.json_data['partitions']` is a list so we can not directly access to a specific partition using its name as an index.

This is what `self.json_data['partitions']` looks like:
`[OrderedDict([('items', [OrderedDict([('name', 'CREATOR_SW_CFG_IMMUTABLE_ROM_EXT_EN'), ('value', '0x000001d4')])]), ('name', 'CREATOR_SW_CFG')])]`

Reply:
ah I see, never mind then.